src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -109,6 +111,9 @@ export function DeterministicLinePlot({
   const { t } = useTranslation()
   const chartRef = React.useRef(null)
   const [enabledPlots, setEnabledPlots] = useState(Object.values(DATA_POINTS))
+  const [rendered, setRendered] = useState(<></>)
+
+  const [debouncedRender] = useDebouncedCallback(() => setRendered(render()), 1000)
```

Comment:
Store the rendered output into the render state. The render function can take upwards of 60msec to finish, so we debounce to a maximum of once every 1000 msec.